### DISCUSSION TITLE
Features missing / ideas

### DISCUSSION THREAD

**Discussion by FoxRunningFromCode** [Akascape/CTkDesigner-Support#12] [Ideas]
Hello. 
While testing out your V4.0, I came across some features that I felt was missing and some widgets i thought would be Nice to have.

- Set background image.
- non user inputable indicator like a dot ● where you can change size and color. Imagine using it for showing boolean signals or perhaps supporting colour input instead.
- Simple static pictures where I can display my Dots ontop of ;)
- Dedicated change page buttons...? (I guess?)

Theese ideas only came to life as im trying to make a userinterface for a program controlling a certain robotic module. So the idea was to have a static picture of the robot, with Dots ● ontop of the specific parts, on the picture, indicating their status.

My apologies if theese features already excist and um just too blind to see it :)

**Comment by Akascape** [comment]
@FoxRunningFromCode 
- You can set background image with CTkLabel or CTkDraw.
- There are page change buttons, you just have to right click on the ctkbutton and select "Add Navigation"

Can you elaborate second and third point? (Any image which shows the thing you want to achieve)

**Comment by FoxRunningFromCode** [comment]
To elaborate on what we're looking at:
This is a quick sketch of the desired Human Machine Interface (HMI) im trying to create. What i feel like im missing is items that the user CANNOT interact with. In this case i want some simple indicator lights (all the red/green dots) that is able to tell the status of a boolean or perhaps im just able to set and switch between x colours, like with the "stacklight".

I dont want the user to be able to interact with the indicators as they should only be used as indicators.
I suppose the radio button does excist but i dont see a way to disable clicking?

![image](https://github.com/Akascape/CTkDesigner-Support/assets/90418563/fd97b9b0-1861-4b5e-9bec-7a9141025548)

For reference, in something like Siemens HMI designer, you have the ability to add small symbols that can change colour depending on the state of the object. The M is added to indicate its a motor. (Perhaps i could just add text "M" ontop of the dot... hmmm)
But the important part being that the user cant interact with the object.

![image](https://github.com/Akascape/CTkDesigner-Support/assets/90418563/1264b971-bf34-4ad5-8681-d13b5503fe0e)

![image](https://github.com/Akascape/CTkDesigner-Support/assets/90418563/5e15d64e-a95c-4a86-8bca-444c14ec2dfc)

**Comment by Akascape** [comment]
@FoxRunningFromCode 
> I suppose the radio button does excist but i dont see a way to disable clicking?

You can disable the clicking by adding `state="readonly"` parameter

Or you can also use CTkMeter to make that dot, if you want to add a letter try using a button and adjust the width, height and corner radius. 
But radiobutton will do the work I guess (make sure to adjust the bg colors as it will be in top of the image)
![image](https://github.com/Akascape/CTkDesigner-Support/assets/89206401/5750d23c-85d4-43f6-b431-79893b95918a)

If you want full transparency in the bg color in top of image, check this simple method: https://github.com/TomSchimansky/CustomTkinter/discussions/2214
